webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -21,26 +23,54 @@ import {getSubscribeModalState, getWebsocketEventState} from 'selectors';
 
 import usePluginApi from 'hooks/usePluginApi';
 import useApiRequestCompletionState from 'hooks/useApiRequestCompletionState';
+
+import {getIncrementedPaginationQueryParamOffset} from 'utils';
 import usePreviousState from 'hooks/usePreviousState';
 
 const ProjectDetails = memo((projectDetails: ProjectDetails) => {
     const {projectName, organizationName} = projectDetails;
 
     // State variables
+    const [paginationQueryParams, setPaginationQueryParams] = useState<PaginationQueryParams>({
+        offset: plugin_constants.common.defaultPageOffset,
+        limit: plugin_constants.common.defaultPageLimit,
+    });
+    const [subscriptionList, setSubscriptionList] = useState<SubscriptionDetails[]>([]);
     const [showAllSubscriptions, setShowAllSubscriptions] = useState(false);
     const [showProjectConfirmationModal, setShowProjectConfirmationModal] = useState(false);
     const [showSubscriptionConfirmationModal, setShowSubscriptionConfirmationModal] = useState(false);
     const [subscriptionToBeDeleted, setSubscriptionToBeDeleted] = useState<SubscriptionPayload>();
+
+    // Hooks
     const {currentChannelId} = useSelector((reduxState: GlobalState) => reduxState.entities.channels);
+    const previousState = usePreviousState({currentChannelId});
+    const dispatch = useDispatch();
+    const {makeApiRequestWithCompletionStatus, getApiState, state} = usePluginApi();
+
     const subscriptionListApiParams = useMemo<FetchSubscriptionList>(() => ({
         project: projectName,
         channel_id: showAllSubscriptions ? '' : currentChannelId,
-    }), [projectName, currentChannelId, showAllSubscriptions]);
+        offset: paginationQueryParams.offset,
+        limit: paginationQueryParams.limit,
+    }), [projectName, currentChannelId, showAllSubscriptions, paginationQueryParams]);
 
-    // Hooks
-    const previousState = usePreviousState({currentChannelId});
-    const dispatch = useDispatch();
-    const {makeApiRequestWithCompletionStatus, makeApiRequest, getApiState, state} = usePluginApi();
+    const {data, isLoading} = getApiState(plugin_constants.pluginApiServiceConfigs.getSubscriptionList.apiServiceName, subscriptionListApiParams);
+    const subscriptionListReturnedByApi = data as SubscriptionDetails[] || [];
```

Comment:
no, because we are not initializing it anywhere else with `[]` so, if the API call is in progress or until we don't have `data` from API it will be `undefined` and doing `...` on `undefined` will lead to error

Reply:
we can keep a check with respect to `isLoading`

Reply:
its the same thing
and its better to always initialise with empty array whenever data is not defined to tackle many cases at once 

checking for loading particularly will not be ideal as, say if API is broken and it's not loading anymore then data could be undefined or null in that case